apis/v1/secretproviderclass_types.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Provider enum for all the provider names
+type Provider string
+
+const (
+	// Azure provider for Azure Key Vault
+	Azure Provider = "Azure"
+	// Vault provider for Hashicorp Vault
+	Vault Provider = "Vault"
```

Comment:
is this file auto-generated? Curious about these constants. Since the driver is written to arbitrary providers it would be best to not have supported providers marked in code